src/framework/ui/uiwidget.cpp

```diff
@@ -1791,6 +1860,18 @@ bool UIWidget::onDoubleClick(const Point& mousePos)
     return callLuaField<bool>("onDoubleClick", mousePos);
 }
 
+UIWidgetPtr UIWidget::getHoveredChild() 
+{
+    for (const UIWidgetPtr& child : m_children) {
+        if (child->isHovered())
+            return child;
+
+        const auto& hovered = child->getHoveredChild();
+        if (hovered) return hovered;
+    }
+    return nullptr;
+}
+
```

Comment:
No for loop needed

you can just use:
g_ui.getHoveredWidget()

then get the id
and then getChildById